app/src/main/java/org/dslul/openboard/inputmethod/latin/settings/UserDictionaryListFragment.java

```diff
@@ -0,0 +1,306 @@
+            localeSet.add(Locale.getDefault().toString());
+        }
+
+        return localeSet;
+    }
+
+    /**
+     * Creates the entries that allow the user to go into the user dictionary for each locale.
+     * @param userDictGroup The group to put the settings in.
+     */
+    protected void createUserDictSettings(final PreferenceGroup userDictGroup) {
+        final SharedPreferences prefs = DeviceProtectedUtils.getSharedPreferences(requireActivity().getApplicationContext());
+        final boolean localeSystemOnly = prefs.getBoolean(Settings.PREF_USE_SYSTEM_LOCALES, true);
+        // List of enabled user dictionary languages
+        final TreeSet<String> enabledUserDictionary = getUserDictionaryLocalesSet(requireActivity());
+        // List of main language
+        final List<InputMethodSubtype> enabledMainSubtype = SubtypeSettingsKt.getEnabledSubtypes(prefs, true);
+        // List of enabled system language
+        final List<Locale> enabledSystemLocale = SubtypeSettingsKt.getSystemLocales();
+        // To combine lists and handle duplicates automatically
+        LinkedHashSet<String> userDictionaryLanguage = new LinkedHashSet<>();
+
+        // Add "For all languages"
+        userDictionaryLanguage.add("");
+
+        // Add the language from the user dictionary if a word is present
+        if (enabledUserDictionary != null) {
+            userDictionaryLanguage.addAll(enabledUserDictionary);
+        }
+
+        // Add the main language selected in the "Language and Layouts" setting
+        for (InputMethodSubtype mainSubtype : enabledMainSubtype) {
+            Locale mainLocale = LocaleUtils.constructLocaleFromString(mainSubtype.getLocale());
+            String mainLocaleString = mainLocale.toString();
+            // Special treatment for the known languages with _ZZ and _zz types
+            if (mainLocaleString.endsWith("_ZZ") || mainLocaleString.endsWith("_zz") || mainLocaleString.equals("zz")) {
+                mainLocaleString = LocaleUtils.getLocaleDisplayNameInSystemLocale(mainLocale, requireActivity()).toLowerCase();
+            }
+            userDictionaryLanguage.add(mainLocaleString);
+
+            // Add secondary language only if main language is selected and "Use system languages" setting is not enabled
+            for (Locale secondSubtype : Settings.getSecondaryLocales(prefs, String.valueOf(mainLocale))) {
+                Locale secondLocale = LocaleUtils.constructLocaleFromString(String.valueOf(secondSubtype));
+                String secondLocaleString = secondLocale.toString();
+                // Special treatment for the known languages with _ZZ and _zz types
+                if (secondLocaleString.endsWith("_ZZ") || secondLocaleString.endsWith("_zz") || secondLocaleString.equals("zz")) {
+                    secondLocaleString = LocaleUtils.getLocaleDisplayNameInSystemLocale(secondLocale, requireActivity()).toLowerCase();
+                }
+                if (!localeSystemOnly) {
```

Comment:
Here you create the locale string, and discard it if `localeSystemOnly`. This seems pretty unnecessary, it looks like you only need this part if `!localeSystemOnly`.

Reply:
Rectified with commit f950744